### DISCUSSION TITLE
How can I assign shaders to objects in the USD stage?

### DISCUSSION THREAD

**Discussion by Elvis33LE** [Autodesk/maya-usd#1314] [Q&A]
Hello there,

I have a pretty simple question. How can I assign shaders to objects in the USD stage?
For example if i load the Attic scene I want to change some shaders but Maya tells me:

# Error: No renderable object is selected for assignment

How does this work?

Thanks in advance.

**Comment by santosd** [comment]
Hi @Elvis33LE,
At this point a material cannot be assigned to a USD prim inside of Maya. The USD prim comes with a material assigned to it programmatically defined in the .usd file. If you would like to assign a different material you would need to bring the prim into Maya through File  > Import (USD Import) to convert the USD data to Maya data and then from there you could assign a different material\texture and export back to USD (USD Export) with the "Materials:" option set to USD Preview Surface. When you create a stage from the new file, the new textures should now be set. I have created and attached a video showing the workflow.
A different way to do this is to edit the texture path in a .usda file. This only works if you have an ascii USD file that you can read in a text editor. Then you would see a section like the following:
```            def Shader "file1"
            {
                uniform token info:id = "UsdUVTexture"
                float4 inputs:fallback = (0.5, 0.5, 0.5, 1)
                asset inputs:file = @C:/Users/Path/To/Texture/File/Mat_Circle_baseColor.jpg@
                float2 inputs:st.connect = </pSphere1/Looks/lambert2SG/file1/TexCoordReader.outputs:result>
                token inputs:wrapS = "repeat"
                token inputs:wrapT = "repeat"
                float3 outputs:rgb

                def Shader "TexCoordReader"
                {
                    uniform token info:id = "UsdPrimvarReader_float2"
                    token inputs:varname.connect = </pSphere1/Looks/lambert2SG.inputs:file1:varname>
                    float2 outputs:result
                }
```
Using a text editor you could edit the `asset inputs:file = @C:/Users/...` section and specify a different path to a texture file and it will change when the usd file is brought into Maya as a stage.
I hope that information is helpful.
-David Santos

https://user-images.githubusercontent.com/29235905/113905930-acd37480-97a1-11eb-9cc7-46a89768fb2f.mp4
